api/common/v1alpha1/ref.go

```diff
@@ -61,6 +63,29 @@ func (in TargetRef) Hash() TargetRefHash {
 	return TargetRefHash(fmt.Sprintf("%s/%s/%s/%s", in.Kind, in.Name, strings.Join(orderedTags, "/"), in.Mesh))
 }
 
+func (in TargetRef) EnvoyTags() (map[string]string, bool) {
```

Comment:
To be honest, I don't see much value in making this a method. TargetRef is a core API object, it doesn't have to "know" how to represent itself in terms of Envoy tags. 

I'd just create a method somewhere under `pkg/xds/`:
```go
func TargetRefToEnvoyTags(tr TargetRef) (map[string]string, bool)
```

Reply:
I think maybe including "Envoy" was unnecessary. There's definitely something inherent to the semantics of the API object here, especially since `MeshServiceSubset` _includes_ tags. This is really just one direction of the isomorphism between `map[string]string`, where `kuma.io/service` is a tag, and `TargetRef`.

Reply:
> I think maybe including "Envoy" was unnecessary. 

How would you call the method then?

Reply:
Moved somewhere else